servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/WriteStreamSubscriber.java

```diff
@@ -366,6 +366,8 @@ void sourceTerminated(@Nullable Throwable cause) {
             }
             this.failureCause = cause;
             setFlag(SOURCE_TERMINATED);
+            // Mark the subscription as CANCELLED to prevent propagating cancel from channelClosed
+            subscriptionUpdater.getAndSet(WriteStreamSubscriber.this, CANCELLED);
```

Comment:
nit: why not just `set` ?

Reply:
This change was extracted to a separate PR #1838 and changed to a simple volatile assignment